### PR TITLE
support menu column

### LINKED ISSUE #15723: Adding context menu column to Declarative table
image

1. The context menu will be a list of options with separators.
2. Each row will have a different set of items. 
3. Some items should also have a link icon to it.

### PR DESCRIPTION
**Pull request by alanrenmsft**
This PR fixes #15723 

image

image

usage:

column definition:
````
{
						displayName: '',
						valueType: azdata.DeclarativeDataType.menu,
						isReadOnly: true,
						width: 30
}
````

data value:

````
						{
							value: {
								commands: ['sqlmigration.openNotebooks',
									['sqlmigration.openNotebooks', 'sqlmigration.openNotebooks'],
									'sqlmigration.openNotebooks'],
								context: 'abc'
							}
						}
`````

### CODE REVIEW

**Review by aasimkhan30 — APPROVED** [review]

File: src/sql/azdata.proposed.d.ts [hunk 1]
```diff
@@ -387,11 +388,23 @@ declare module 'azdata' {
...
+
+	export interface DeclarativeTableMenuCellValue {
+		/**
+		 * commands for the menu. Use an array for a group and menu separators will be added.
+		 */
+		commands: (string | string[])[];
+		/**
+		 * context that will be passed to the commands.
+		 */
+		context: any
```
Comment (Charles-Gagnon) [inline src/sql/azdata.proposed.d.ts:400]: Not sure we want to allow passing anything as the context - there's tons of stuff that will error out when it tries to go over the JSON RPC and we should at least try to tell users what stuff is safe to send over. 

We should have a discussion about what this "context" entails, but could we not just have a simple key:value mapping to pass over or something?

File: src/sql/workbench/browser/modelComponents/media/declarativeTable.css [hunk 2]
```diff
@@ -26,3 +26,14 @@
 	width: 100%;
 	text-align: center;
 }
+
+.declarative-table .toggle-more {
+	background-size: 16px !important;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/browser/modelComponents/media/declarativeTable.css:31]: What are you overriding here that requires the important?

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]

File: src/sql/azdata.proposed.d.ts (hunk 1 above)
Comment (alanrenmsft) [reply src/sql/azdata.proposed.d.ts:400]: yeah, let me restrict the supported types here.

File: src/sql/workbench/browser/modelComponents/media/declarativeTable.css (hunk 2 above)
Comment (alanrenmsft) [reply src/sql/workbench/browser/modelComponents/media/declarativeTable.css]: I don't actually need that, also as a good practice, I am adding more specific selectors.

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**